types/gulp-stylus/index.d.ts

```diff
@@ -0,0 +1,24 @@
+// Type definitions for gulp-stylus 2.7
+// Project: http://github.com/stevelacy/gulp-stylus
+// Definitions by: Takesi Tokugawa <https://github.com/TokugawaTakesi>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node"/>
+
+interface GulpStylus {
+    (options?: GulpStylus.Options): NodeJS.ReadWriteStream;
+}
+
+declare namespace GulpStylus {
+    interface Options {
+        compress?: boolean;
+        linenos?: boolean;
+        'include css'?: boolean;
+        rawDefine?: {
+            data: any
+        };
+    }
+}
+
+declare var gulpStylus: GulpStylus;
```

Comment:
Rather than using an interface for this, I'd suggest something like the following.
```ts
function gulpStylus(options?: GulpStylus.Options): NodeJS.ReadWriteStream;

declare namespace gulpStylus {
...
}

export = gulpStylus;
```

This makes it clear that the library exports a function, but also the type `Options`.